buildingmotif/database/tables.py

```diff
@@ -138,3 +139,20 @@ class DBTemplate(Base):
             name="name_library_unique_constraint",
         ),
     )
+
+
+@event.listens_for(DBTemplate, "after_delete")
+def handle_template_deletion(mapper, connection, target):
+    """Event listener to handle deletion of a template and its dependents."""
+    from buildingmotif import get_building_motif
+
+    session = get_building_motif().session
+
+    # Identify all dependant templates to delete and delete them (and their libraries).
+    delete_libraries = []
+    dependants = target.dependants[:]
+    for dependant in dependants:
+        session.delete(dependant)
```

Comment:
based on the test output it seems like sqlalchemy doesn't like `session.delete` in event functions